src/megaapi_impl.cpp

```diff
@@ -24415,30 +24417,36 @@ MegaSyncPrivate::MegaSyncPrivate(const char *path, const char *name, handle node
     this->regExp = NULL;
 }
 
-MegaSyncPrivate::MegaSyncPrivate(const SyncConfig& config, Sync* syncPtr)
+MegaSyncPrivate::MegaSyncPrivate(const SyncConfig& config, Sync* syncPtr, MegaClient* client)
     : mType(static_cast<SyncType>(config.getType()))
     , mActive(syncPtr && syncPtr->state >= 0)
     , mEnabled(config.getEnabled())
 {
     this->megaHandle = config.getRemoteNode();
     this->localFolder = NULL;
-    setLocalFolder(config.getLocalPath().c_str());
+    setLocalFolder(config.getLocalPath().toPath(*client->fsaccess).c_str());
     this->mName= NULL;
-    if (!config.getName().empty())
+    if (!config.mName.empty())
     {
-        setName(config.getName().c_str());
+        setName(config.mName.c_str());
     }
     else
     {
         //using localpath as name:
-        setName(config.getLocalPath().c_str());
+        setName(config.getLocalPath().toPath(*client->fsaccess).c_str());
     }
     this->megaFolder = NULL;
     this->fingerprint = 0;
     this->regExp = NULL;
 
     setLocalFingerprint(static_cast<long long>(config.getLocalFingerprint()));
-    setMegaFolder(config.getRemotePath().c_str());
+
+    // per the spec in megaapi.h:  `Get the path of the remote folder that is being synced`
+    if (auto n = client->nodebyhandle(config.getRemoteNode()))
+    {
+        setMegaFolder(n->displaypath().c_str());
+    }
```

Comment:
I believe this change has a bad side effect. MEGAsync expects the last known path of the remote folder to be displayed in case the node was deleted. If we do this, we will return an empty path for deleted remotes.

Reply:
checked, it even triggers assert:
```
        // Col 2: Mega Folder
        QSyncItemWidget *megaFolder = new QSyncItemWidget();
        assert(syncSetting->getMegaFolder().size() && "remote folder lacks path"); <- this one
        megaFolder->setPathAndGuessName(syncSetting->getMegaFolder().size()?syncSetting->getMegaFolder():QString::fromUtf8("---"));
        megaFolder->setToolTip(syncSetting->getMegaFolder());
        ui->tSyncs->setCellWidget(i, 1, megaFolder);
```

Reply:
Ah, in that case we should change the documentation (and maybe the name too, to alert downstream clients) then, and make it clear that the path is not the one matching the node anymore, but rather a copy from when the sync was last executing.

Reply:
This one is resolved - the functionality is the same but the name is updated to make it clear that the node handle is the definitive member, and this one is just a snapshot from some time back:   getLastKnownMegaFolder